frontend/src/components/cases/case-attachments-section.tsx

```diff
@@ -0,0 +1,735 @@
+  return "text-gray-600 bg-gray-50"
+}
+
+function formatFileSize(bytes: number): string {
+  if (bytes === 0) return "0 Bytes"
+  const k = 1024
+  const sizes = ["Bytes", "KB", "MB", "GB"]
+  const i = Math.floor(Math.log(bytes) / Math.log(k))
+  return Number.parseFloat((bytes / Math.pow(k, i)).toFixed(1)) + " " + sizes[i]
+}
+
+function truncateHash(hash: string): string {
+  return hash.substring(0, 8)
+}
+
+function getUploaderName(creatorId: string | null | undefined): string {
+  if (!creatorId) return "Unknown"
+  return "User"
+}
+
+// Add constant for max file size (20MB in bytes)
+const MAX_FILE_SIZE_BYTES = 20 * 1024 * 1024 // 20MB
+
+export function CaseAttachmentsSection({
+  caseId,
+  workspaceId,
+}: CaseAttachmentsSectionProps) {
+  const [isUploading, setIsUploading] = useState(false)
+  const [isDragOver, setIsDragOver] = useState(false)
+  const [previewAttachment, setPreviewAttachment] =
+    useState<CaseAttachmentRead | null>(null)
+  const [previewImageUrl, setPreviewImageUrl] = useState<string | null>(null)
+  const fileInputRef = useRef<HTMLInputElement>(null)
+  const queryClient = useQueryClient()
+
+  // Fetch attachments from API
+  const {
+    data: attachments = [],
+    isLoading: attachmentsLoading,
+    error: attachmentsError,
+  } = useQuery<CaseAttachmentRead[], ApiError>({
+    queryKey: ["case-attachments", caseId, workspaceId],
+    queryFn: async () => await casesListAttachments({ caseId, workspaceId }),
+  })
+
+  // Upload attachment mutation
+  const uploadMutation = useMutation({
+    mutationFn: async (file: File) => {
+      return await casesCreateAttachment({
+        caseId,
+        workspaceId,
+        formData: {
+          file,
+        },
+      })
+    },
+    onSuccess: (data, file) => {
+      queryClient.invalidateQueries({
+        queryKey: ["case-attachments", caseId, workspaceId],
+      })
+      queryClient.invalidateQueries({
+        queryKey: ["case-events", caseId, workspaceId],
+      })
+      setIsUploading(false)
+      toast({
+        title: "Attachment uploaded successfully",
+        description: `${file.name} has been added to the case`,
+      })
+    },
+    onError: (error: ApiError, file) => {
+      console.error("Failed to upload attachment:", error)
+      setIsUploading(false)
+
+      // Handle structured error responses with specific HTTP status codes
+      if (
+        error.status === 415 &&
+        error.body &&
+        typeof error.body === "object"
+      ) {
+        const body = error.body as ApiErrorBody
+        const detail = body.detail
+
+        if (
+          typeof detail === "object" &&
+          detail?.error === "unsupported_file_extension"
+        ) {
+          toast({
+            title: "File type not supported",
+            description: `${file.name} cannot be uploaded. Allowed file types: ${detail.allowed_extensions?.join(", ") || "txt, pdf, png, jpeg, gif, csv"}`,
+          })
+          return
+        }
```

Comment:
ideally we autogenerate the types from the backend openapi spec, then use type guards to narrow the types into their expected types